packages/runtime/test-runtime-utils/src/mockStorage.ts

```diff
@@ -17,41 +18,39 @@ export class MockStorage implements IChannelStorageService {
         return new MockStorage(tree);
     }
 
-    private static readCore(tree: ITree, paths: string[]): string {
+    public async readBlob(path: string): Promise<ArrayBufferLike> {
+        const blob = await this.readBlobInternal(this.tree, path.split("/"));
+        return toBuffer(blob.contents, blob.encoding);
+    }
+
+    constructor(protected tree?: ITree) {
+    }
+
+    public async contains(path: string): Promise<boolean> {
+        return await this.readBlob(path) !== undefined;
```

Comment:
I do not think this code is right. I believe before your changes, it was using readCore, that could return undefined.
Now, readBlob() will never return undefined. It will throw (trying to do blob.contents when blob === undefined).
So you nedd to call directly readBlobInternal here.
Also it would be great to add assert in readBlob() before we do blob.contents  with more meaningful message, such that when this condition happens, it's easy to debug it, not chase through the code where we unreferenced undefined.

I have more general ask: could you please inspect all the refactoring changes for keeping exactly same behaviors? While there are some number of comments I've left about existing code being incorrect (and thus ideally us fixing it as we go through it), the general philosophy of safe refactoring (and that's pretty much what you do here) is to not change behavior a single bit. I.e. with each step, analyze what existing code does, what are the interesting corner cases and preserve them one to one.
While we should also access feasibility to add unit tests for all the code you are changing (and ideally merge UTs before you do refactoring, such that we can validate that refactoring did not break newly added tests), it's also very important not to fully blindly rely on tests (and we quite often have no good coverage) and very carefully work with existing code not to change its behaviors.
Smaller steps (and smaller PRs) with follow up issues works the best.